ledger-core/v2/runner/eventsink.go

```diff
@@ -6,53 +6,47 @@
 package runner
 
 import (
-	"github.com/pkg/errors"
-
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner/call"
 	"github.com/insolar/assured-ledger/ledger-core/v2/runner/execution"
 	descriptor "github.com/insolar/assured-ledger/ledger-core/v2/runner/executionevent"
 	"github.com/insolar/assured-ledger/ledger-core/v2/runner/executionupdate"
 	"github.com/insolar/assured-ledger/ledger-core/v2/runner/requestresult"
 )
 
 type executionEventSink struct {
+	id      call.ID
 	context execution.Context
 	output  chan *executionupdate.ContractExecutionStateUpdate
```

Comment:
You don't need to use channel output - a direct pointer to RunState will be fine. There is no racing here, as RunState will only be accessed after triggering a callback.
Same for input - this is only one time assign and one time read.

Reply:
Channel for input (outgoing result) is needed to wakeup goroutine that executes contract